app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file Button.tsx
+ *
+ * Button component
+ */
+import clsx from 'clsx'
+import * as reactAriaComponents from 'react-aria-components'
+import * as tailwindMerge from 'tailwind-merge'
+
+import SvgMask from '#/components/SvgMask'
+
+/**
+ * Props for the Button component
+ */
+export interface ButtonProps extends reactAriaComponents.ButtonProps {
+  readonly variant: 'icon' | 'primary'
+  readonly icon?: string
+  /**
+   * FIXME: This is not yet implemented
+   * The position of the icon in the button
+   * @default 'start'
+   */
+  readonly iconPosition?: 'end' | 'start'
+}
+
+const DEFAULT_CLASSES =
+  'flex cursor-pointer rounded-sm border border-transparent transition-opacity duration-200 ease-in-out'
+const FOCUS_CLASSES =
+  'focus-visible:outline-offset-2 focus:outline-none focus-visible:outline focus-visible:outline-blue-500 '
+const ICON_CLASSES = 'opacity-50 hover:opacity-100'
+const EXTRA_CLICK_ZONE_CLASSES =
+  'flex relative before:inset-[-12px] before:absolute before:z-10 before:content-""'
+// not yet implemented
+const PRIMARY_CLASSES = ''
+const DISABLED_CLASSES = 'disabled:opacity-50 disabled:cursor-not-allowed'
+
+/**
+ * A button allows a user to perform an action, with mouse, touch, and keyboard interactions.
+ */
+export function Button(props: ButtonProps) {
+  const { variant, className, children, icon, ...ariaButtonProps } = props
+
+  const classes = clsx(DEFAULT_CLASSES, DISABLED_CLASSES, FOCUS_CLASSES, {
+    [ICON_CLASSES]: variant === 'icon',
```

Comment:
Sorry, could you elaborate? I didn't get why I should use `ariant === 'icon' && ICON_CLASSES`

Reply:
well, is there a reason to *not* use it? imo:
- we should at least be consistent in the call. either everything should be in the object or nothing should be
- `[ICON_CLASSES]: variant === 'icon'` is not as intuitive (doesn't read as naturally) as `variant === 'icon' && ICON_CLASSES` imo.

Reply:
If you don't like it, it doesn't mean it's bad. And it's absolutely _natural_: https://github.com/lukeed/clsx?tab=readme-ov-file#usage

Reply:
it's not *natural*, it's *a usecase they support*. because they support mixing and matching of every input format. that doesn't necessarily mean we want to randomly mix and match input formats, however.

Reply:
The same way it doesn't mean that it's bad though.